Broken or Flaky test: ScanServerMetadataEntriesIT

**Test name(s)**
 - org.apache.accumulo.test.ScanServerMetadataEntriesIT.testGcRunScanServerReferences
 - org.apache.accumulo.test.ScanServerMetadataEntriesIT.testBatchScanServerMetadataEntries
 - org.apache.accumulo.test.ScanServerMetadataEntriesIT.testServerContextMethods (sometimes)
 
**Describe the failure observed**
```
[INFO] Running org.apache.accumulo.test.ScanServerMetadataEntriesIT
[ERROR] Tests run: 4, Failures: 3, Errors: 0, Skipped: 0, Time elapsed: 36.34 s <<< FAILURE! -- in org.apache.accumulo.test.ScanServerMetadataEntriesIT
[ERROR] org.apache.accumulo.test.ScanServerMetadataEntriesIT.testGcRunScanServerReferences -- Time elapsed: 1.978 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <3> but was: <1>
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:531)
	at org.apache.accumulo.test.ScanServerMetadataEntriesIT.testGcRunScanServerReferences(ScanServerMetadataEntriesIT.java:248)

[ERROR] org.apache.accumulo.test.ScanServerMetadataEntriesIT.testServerContextMethods -- Time elapsed: 0.017 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <2> but was: <5>
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:632)
	at org.apache.accumulo.test.ScanServerMetadataEntriesIT.testServerContextMethods(ScanServerMetadataEntriesIT.java:121)

[ERROR] org.apache.accumulo.test.ScanServerMetadataEntriesIT.testBatchScanServerMetadataEntries -- Time elapsed: 2.027 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <3> but was: <8>
	at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:632)
	at org.apache.accumulo.test.ScanServerMetadataEntriesIT.testBatchScanServerMetadataEntries(ScanServerMetadataEntriesIT.java:198)
```

**Testing Environment:**
 - Version of this project: 3.1.0-SNAPSHOT
 - First commit known to fail (or current commit): 15035d7b2705b606e401a09984e7694cb263e274
```
Executing Maven:  -B -f /var/lib/jenkins/workspace/3.1/pom.xml -V verify -Dfailsafe.rerunFailingTestsCount=5
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /var/lib/jenkins/tools/hudson.tasks.Maven_MavenInstallation/Maven_3.9.6
Java version: 17.0.12, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-17-openjdk-17.0.12.0.7-2.fc40.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.9.9-200.fc40.x86_64", arch: "amd64", family: "unix"
```

**What have you tried already?**
Re-ran test a few times locally and in Jenkins. It pretty reliably fails.

**Additional context**
The commit that caused the failure was PR #4691

--- a/test/src/main/java/org/apache/accumulo/test/ScanServerMetadataEntriesIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/ScanServerMetadataEntriesIT.java
@@ -243,8 +243,7 @@ public class ScanServerMetadataEntriesIT extends SharedMiniClusterBase {
 
       Set<String> metadataScanFileRefs = new HashSet<>();
       metadataEntries.forEach(m -> {
-        String file = m.getKey().getColumnQualifier().toString();
-        metadataScanFileRefs.add(file);
+        metadataScanFileRefs.add(new ScanServerRefTabletFile(m.getKey()).getFilePath().toString());
       });
       assertEquals(fileCount, metadataScanFileRefs.size());
 

--- a/test/src/main/java/org/apache/accumulo/test/ScanServerMetadataEntriesIT.java
+++ b/test/src/main/java/org/apache/accumulo/test/ScanServerMetadataEntriesIT.java
@@ -231,43 +231,42 @@ public class ScanServerMetadataEntriesIT extends SharedMiniClusterBase {
         Iterator<Entry<Key,Value>> iter = scanner.iterator();
         assertTrue(iter.hasNext());
         assertNotNull(iter.next());
-
-        List<Entry<Key,Value>> metadataEntries = null;
-        try (Scanner scanner2 =
-            client.createScanner(AccumuloTable.SCAN_REF.tableName(), Authorizations.EMPTY)) {
-          metadataEntries = scanner2.stream().distinct().collect(Collectors.toList());
-        }
-        assertEquals(fileCount, metadataEntries.size());
-        metadataEntries.forEach(e -> log.info("{}", e.getKey()));
-
-        Set<String> metadataScanFileRefs = new HashSet<>();
-        metadataEntries.forEach(m -> {
-          String file = m.getKey().getRow().toString();
-          metadataScanFileRefs.add(file);
-        });
-        assertEquals(fileCount, metadataScanFileRefs.size());
-
-        assertEquals(fileCount, ctx.getAmple().scanServerRefs().list().count());
-        List<Reference> refs;
-        try (Stream<Reference> references = gc.getReferences()) {
-          refs = references.collect(Collectors.toList());
-        }
-        assertTrue(refs.size() > fileCount * 2);
-        List<Reference> tableRefs =
-            refs.stream().filter(r -> r.getTableId().equals(tid) && !r.isDirectory())
-                .peek(r -> assertTrue(metadataScanFileRefs.contains(r.getMetadataPath())))
-                .collect(Collectors.toList());
-        log.info("Reference List:{}", tableRefs);
-        // There should be 6 references here. 3 for the table file entries, and 3 for the scan
-        // server references
-        assertEquals(fileCount * 2, tableRefs.size());
-
-        Set<String> deduplicatedReferences =
-            tableRefs.stream().map(Reference::getMetadataPath).collect(Collectors.toSet());
-
-        assertEquals(fileCount, deduplicatedReferences.size());
       }
 
+      List<Entry<Key,Value>> metadataEntries = null;
+      try (Scanner scanner2 =
+          client.createScanner(AccumuloTable.SCAN_REF.tableName(), Authorizations.EMPTY)) {
+        metadataEntries = scanner2.stream().distinct().collect(Collectors.toList());
+      }
+      assertEquals(fileCount, metadataEntries.size());
+      metadataEntries.forEach(e -> log.info("{}", e.getKey()));
+
+      Set<String> metadataScanFileRefs = new HashSet<>();
+      metadataEntries.forEach(m -> {
+        String file = m.getKey().getColumnQualifier().toString();
+        metadataScanFileRefs.add(file);
+      });
+      assertEquals(fileCount, metadataScanFileRefs.size());
+
+      assertEquals(fileCount, ctx.getAmple().scanServerRefs().list().count());
+      List<Reference> refs;
+      try (Stream<Reference> references = gc.getReferences()) {
+        refs = references.collect(Collectors.toList());
+      }
+      assertTrue(refs.size() > fileCount * 2);
+      List<Reference> tableRefs =
+          refs.stream().filter(r -> r.getTableId().equals(tid) && !r.isDirectory())
+              .peek(r -> assertTrue(metadataScanFileRefs.contains(r.getMetadataPath())))
+              .collect(Collectors.toList());
+      log.info("Reference List:{}", tableRefs);
+      // There should be 6 references here. 3 for the table file entries, and 3 for the scan
+      // server references
+      assertEquals(fileCount * 2, tableRefs.size());
+
+      Set<String> deduplicatedReferences =
+          tableRefs.stream().map(Reference::getMetadataPath).collect(Collectors.toSet());
+
+      assertEquals(fileCount, deduplicatedReferences.size());
       client.tableOperations().delete(tableName);
     }
     // close happens asynchronously. Let the test fail by timeout

